robot/Cumulus/resources/RecurringDonationsPageObject.py

```diff
@@ -18,10 +18,16 @@ class RDListingPage(BaseNPSPPage, ListingPage):
     @capture_screenshot_on_error
     def wait_for_rd2_modal(self):
         """Based on the button name (Cancel)  or (Save) on the modal footer, selects and clicks on the respective button"""
-        self.builtin.sleep(2,"Wait Needed for now to wait for the new modal")
+        self.builtin.sleep(3,"Wait Needed for now to wait for the new modal")
         btnlocator = npsp_lex_locators["button-with-text"].format("Save")
         self.selenium.scroll_element_into_view(btnlocator)
         self.selenium.wait_until_element_is_visible(btnlocator,60)
+        for i in range(3):
```

Comment:
Why is this necessary, since you're already waiting up to 60 seconds for the element to be visible?

Reply:
This logic is added to help handle the unpredicted load times of the Recurring donations modal.

Reply:
I still don't quite understand. You're already waiting for it to be visible. Are you saying that the element could be visible but that `self.npsp.check_if_element_displayed(btnlocator)` can still return `False`? What else is `check_if_element_displayed` doing besides checking that it is visible?

If this extra check really is necessary, please add a comment which explains why it is necessary. Otherwise, someone else may come by later and think it's a redundant check and remove it.

Reply:
Removed the redundant wait and added a comment for the logic I added in the loop